xdsl/dialects/mpi.py

```diff
@@ -352,4 +356,461 @@ class Finalize(MPIBaseOp):
     Init,
     Finalize,
     CommRank,
-], [RequestType, StatusType])
+], [
+    RequestType,
+    StatusType,
+])
+
+
+@dataclasses.dataclass
+class MpiLibraryInfo:
+    """
+    This object is meant to capture characteristics of a specific MPI implementations.
+
+    It holds magic values, sizes of structs, field offsets and much more.
+
+    We need these as we currently cannot load these library headers into the programs we want to lower,
+    therefore we need to generate our own external stubs and load magic values directly.
+
+    This way of doing it is inherently fragile, but we don't know of any better way.
+    We plan to include a C file that automagically extracts all this information from MPI headers.
+
+    These defaults have been chosen to work with **our** version of OpenMPI. No guarantees of portability!
```

Comment:
Please specify version

Reply:
Will do!

In the future I want to supply a C file that generates this information later on so that people can use their own MPI libraries!